### ISSUE TITLE
[Clarity] Introduce fixed size array type.

### ISSUE THREAD

**Issue by LNow** [stacks-network/stacks-blockchain#3057]
**Is your feature request related to a problem? Please describe.**
In order to access specific element of a sequence (list, string, buff) we have to use following:
```clarity
(element-at sequence index)
```
and then unwrap the result and handle potential `none`.
This is fine for code when we work with sequences of specific length that can also be smaller than limit we defined. However there are use-cases where we work with fixed size sequences that can't be smaller than certain limit, because they are fixed. For them the need to unwrap result of `element-at` is cumbersome.

For example `(sha256 value)` always produce `(buff 32)`. There is no way that it could produce `(buff 3)` or `(buff 31)` it is always `(buff 32)`.
Despite that when I want to access first byte of its result I have to do `(unwrap-panic (element-at sha256result u0))`

It not only makes code less readable, but increase execution costs.

**Describe the solution you'd like**
I'd like to have new fixed size array type, with fast get/set methods that panics when called with index that is out of bound.

Type declaration: `(array len entry-type)`
Example: 
```clarity
(define-data-var my-array (array 5 uint) (array u1 u2 u3 u4 u5))
(define-data-var other-array (array 10 uint) (array u1 u2)) ;; this should fail during code check because initial value is to short
```

Extracting element: `(array-get array index)`
Returns element stored under specified index or panics when index is out of bound.
```clarity
(array-get (array u1 u2 u3) u2) ;; returns u3
(array-get (array "hello" "world") u2) ;; panics with error index out of bound 
```

Modifying element: `(array-set array index new-value)`
```clarity
(array-set (array u1 u2 u3) u2 u200) ;; returns (array u1 u2 u200)
(array-set (array "hello" "world") u2 "universe") ;; panics with error index out of bound 
```

**Comment by jcnelson** [comment]
I think `unwrap-panic` is one of the cheapest operations?  While the extra characters aren't free (since contract space isn't free), how much cost is being eaten up by it?

**Comment by LNow** [comment]
Yes, it is one of the cheapest functions. Main reason why I would like to have fixed size arrays is to improve code readability and maintainability.

I don't like cluttering code with `unwrap-panic` especially when I know there is no way to reach panic execution path unless I forgot to check function inputs.

`element-at-panic` that works and is exactly as expensive as `element-at` + `unwrap-panic` would also do the job. 

Fixed size array would bring one additional benefit - when used as function argument they would clearly indicate that passing anything shorter will lead to immanent execution panic. With sequences you have to read and understand the code to know that.
